src/backend/joanie/core/api/client.py

```diff
@@ -1116,26 +1116,29 @@ def download(self, request, pk=None):  # pylint: disable=unused-argument, invali
                 "Cannot get contract when an order is not yet validated."
             )
 
-        if contract.submitted_for_signature_on:
+        was_fully_signed = (
+            contract.student_signed_on
+            and contract.organization_signed_on
+            and contract.submitted_for_signature_on is None
+        )
+
+        if not was_fully_signed:
             raise ValidationError(
                 "Cannot download a contract when it is not yet fully signed."
             )
 
-        context_data = contract_definition.generate_document_context(
-            contract_definition=contract.definition,
-            user=self.request.user,
-            order=contract.order,
-        )
-        contract_definition_pdf_bytes = issuers.generate_document(
-            contract.definition.name, context_data
+        signed_contract_pdf_bytes = (
+            contract_utility.get_signed_contract_pdf_bytes_from_provider(
+                signature_backend_reference=contract.signature_backend_reference
+            )
```

Comment:
We could reuse the method `get_pdf_bytes_of_contracts` of the `Contract` model instead of create a new util.

IMO, with the util `get_signed_contract_pdf_bytes_from_provider` you are duplicating code. You should at least prevent this code duplication between `get_signed_contract_pdf_bytes_from_provider` and `Contract.get_pdf_bytes_of_contracts`.